src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -2459,6 +2459,69 @@ public static void Deconstruct(this C value, out int a, out string b)
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
```

Comment:
I wrote these tests to confirm the behavior for matching patterns via extensions with ref/in. However they're useful tests so I left them in.